world/shared_task_manager.cpp

```diff
@@ -0,0 +1,1759 @@
+#include "shared_task_manager.h"
+#include "cliententry.h"
+#include "clientlist.h"
+#include "dynamic_zone.h"
+#include "dynamic_zone_manager.h"
+#include "zonelist.h"
+#include "zoneserver.h"
+#include "shared_task_world_messaging.h"
+#include "../common/repositories/character_data_repository.h"
+#include "../common/repositories/character_task_timers_repository.h"
+#include "../common/repositories/shared_task_members_repository.h"
+#include "../common/repositories/shared_task_activity_state_repository.h"
+#include "../common/repositories/completed_shared_tasks_repository.h"
+#include "../common/repositories/completed_shared_task_members_repository.h"
+#include "../common/repositories/completed_shared_task_activity_state_repository.h"
+#include "../common/repositories/shared_task_dynamic_zones_repository.h"
+#include <ctime>
+
+extern ClientList client_list;
+extern ZSList     zoneserver_list;
+
+SharedTaskManager *SharedTaskManager::SetDatabase(Database *db)
+{
+	SharedTaskManager::m_database = db;
+
+	return this;
+}
+
+SharedTaskManager *SharedTaskManager::SetContentDatabase(Database *db)
+{
+	SharedTaskManager::m_content_database = db;
+
+	return this;
+}
+
+std::vector<SharedTaskMember> SharedTaskManager::GetRequestMembers(
+	uint32 requestor_character_id,
+	const std::vector<CharacterDataRepository::CharacterData>& characters
+)
+{
+	std::vector<SharedTaskMember> request_members = {};
+	request_members.reserve(characters.size());
+
+	for (const auto& character : characters) {
+		SharedTaskMember member = {};
+		member.character_id   = character.id;
+		member.character_name = character.name;
+
+		// if the solo/raid/group member is a leader, make sure we tag it as such
+		if (character.id == requestor_character_id) {
+			member.is_leader = true;
+		}
+
+		request_members.emplace_back(member);
+	}
+
+	return request_members;
+}
+
+void SharedTaskManager::AttemptSharedTaskCreation(
+	uint32 requested_task_id,
+	uint32 requested_character_id,
+	uint32 npc_type_id
+)
+{
+	auto task = GetSharedTaskDataByTaskId(requested_task_id);
+	if (task.id != 0 && task.type == TASK_TYPE_SHARED) {
+		LogTasksDetail(
+			"[AttemptSharedTaskCreation] Found Shared Task ({}) [{}]",
+			requested_task_id,
+			task.title
+		);
+	}
+
+	// shared task validation needs character names and levels (not stored in SharedTaskMember)
```

Comment:
Yea, commented is a little outdated regarding names. It's probably fine since levels aren't stored and it describes what the validation needs but feel free to remove the last part